controllers/placementapi_controller.go

```diff
@@ -242,6 +243,11 @@ func (r *PlacementAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		}
 	}()
 
+	// Init Topology condition if there's a reference
+	if instance.Spec.TopologyRef != nil {
+		c := condition.UnknownCondition(condition.TopologyReadyCondition, condition.InitReason, condition.TopologyReadyInitMessage)
+		savedConditions.Set(c)
+	}
```

Comment:
please move the condition initialization to initConditions() where the rest of the conditions are initialized. Also I suggest to avoid the conditional condition initialization and instead always have the condition added to the list and ensureDeployment() should set the condition ready / error depending only on the result of ensurePlacementAPITopology(). This way all the repeated checking of `instance.Spec.TopologyRef != nil' can be encapsulated in ensurePlacementAPITopology

Reply:
Ok sorry, it makes sense. I'm going to move it under `initConditions`. My first implementation had `TopologyReadyCondition` always defined (with a simpler logic), following the same lifecycle of the other conditions.
I think there was an agreement to not see this at all if no `TopologyRef` is passed, hence the conditional, which is evaluated only when the `.Spec` contains a reference to a `Topology`. 
As I mentioned, I changed the implementation based on other suggestions, but I understand you might not like this approach, and if it is a blocker for placement I can change the implementation (even though it will slightly diverge from the others).

Reply:
Lets keep the condition conditional for consistency sake. Maybe after the deadlines we can re-discuss if we want to change it to non-conditional everywhere.